salt/orchestrate/edx/services/mongodb.sls

```diff
@@ -43,12 +47,34 @@ deploy_mongodb_cloud_map:
     - require:
         - file: generate_mongodb_cloud_map_file
 
-resize_root_device_to_use_full_disk:
-  salt.state:
+format_data_drive:
+  salt.function:
+    - tgt: 'G@roles:mongodb and G@environment:{{ ENVIRONMENT }}'
+    - tgt_type: compound
+    - name: state.single
+    - arg:
+        - blockdev.formatted
+    - kwarg:
+        name: /dev/xvdb
+        fs_type: ext4
+    - require:
+        - salt: deploy_mongodb_cloud_map
+
+mount_data_drive:
+  salt.function:
     - tgt: 'G@roles:mongodb and G@environment:{{ ENVIRONMENT }}'
     - tgt_type: compound
-    - sls:
-        - utils.grow_partition
+    - name: state.single
+    - arg:
+        - mount.mounted
+    - kwarg:
+        name: /var/lib/mongodb
+        device: /dev/xvdb
+        fstype: ext4
+        mkmnt: True
+        opts: 'relatime,user'
```

Comment:
I think it's more about adding the `user` option and if you specify anything then you also need to add `relatime`.